File: isabelle.eclipse.launch/src/isabelle/eclipse/launch/IsabelleLaunchImages.java
package isabelle.eclipse.launch;

import java.net.MalformedURLException;
import java.net.URL;

import org.eclipse.jface.resource.ImageDescriptor;
import org.eclipse.jface.resource.ImageRegistry;
import org.eclipse.swt.graphics.Image;


public class IsabelleLaunchImages {

	/* Declare Common paths */
	private static URL ICON_BASE_URL= null;

	static {
		String pathSuffix = "icons/"; //$NON-NLS-1$	
		ICON_BASE_URL= IsabelleLaunchPlugin.getDefault().getBundle().getEntry(pathSuffix);
	}
	
	public static final String IMG_TAB_MAIN = ICON_BASE_URL + "main_tab.gif";
	public static final String IMG_TAB_INSTALLATION = ICON_BASE_URL + "isabelle.png";
	public static final String IMG_DRIVER = ICON_BASE_URL + "isabelle.png";
	public static final String IMG_LOGIC = ICON_BASE_URL + "logic_obj.gif";
	
	/**
	 * Returns the <code>Image<code> identified by the given path,
	 * or <code>null</code> if it does not exist.
	 */
	public static Image getImage(String path) {
		ImageRegistry imageRegistry = IsabelleLaunchPlugin.getDefault().getImageRegistry();
		Image image = imageRegistry.get(path);
		if (image == null) {
			ImageDescriptor desc = ImageDescriptor.getMissingImageDescriptor();
			try {
				desc = ImageDescriptor.createFromURL(new URL(path));
				imageRegistry.put(path, desc);
				image = imageRegistry.get(path);
			} catch (MalformedURLException me) {
			}
		}
		
		return image;
	}
	
}

File: isabelle.eclipse.launch/src/isabelle/eclipse/launch/IsabelleLaunchImages.scala
package isabelle.eclipse.launch

import java.net.{MalformedURLException, URL}

import org.eclipse.jface.resource.ImageDescriptor


/**
 * Isabelle launch image definitions.
 *
 * When images are used in label providers (e.g. where Image) is required, they must be disposed manually.
 * For convenience, [[org.eclipse.jface.resource.ResourceManager]] could be used.
 *
 * @author Andrius Velykis
 */
object IsabelleLaunchImages {

  private lazy val ICON_BASE_URL = IsabelleLaunchPlugin.plugin.getBundle.getEntry("icons/")

  val MISSING_ICON = ImageDescriptor.getMissingImageDescriptor

  lazy val TAB_MAIN = create("main_tab.gif")
  lazy val TAB_INSTALLATION = create("isabelle.png")
  lazy val TAB_SESSION_DIRS = create("session-dirs.gif")

  lazy val SESSION = create("logic_obj.gif")

  private def create(iconPath: String) = {
    try {
      val url = new URL(ICON_BASE_URL, iconPath)
      ImageDescriptor.createFromURL(url)
    } catch {
      case _: MalformedURLException => MISSING_ICON
    }
  }

}
